File: frontends/Matlab/Gencode/runcode.m
function runstr = runcode(pde, numpde, mpiprocs)

if nargin<3
  mpiprocs = pde.mpiprocs;
end

disp("Run C++ Exasim code ...")
cdir = pwd();
cd(char(pde.buildpath));

DataPath = pde.buildpath;

mystr = " " + num2str(numpde) + " ";
if numpde>100 % two-domain problems
  numpde = 2;
end
if numpde==1
    mystr = mystr + DataPath + "/datain/ " + DataPath + "/dataout/out";
else    
    for i = 1:numpde
        mystr = mystr + DataPath + "/datain" + num2str(i) + "/ " + DataPath + "/dataout" + num2str(i) + "/out";
        mystr = mystr + " ";
    end    
end

mpirun = pde.mpirun;
if pde.platform == "cpu"        
    if mpiprocs==1        
        exec = "./cpuEXASIM ";        
        runstr = "!" + exec + mystr;
    else        
        exec = " ./cpumpiEXASIM ";
        runstr = "!" + mpirun + " -np " + string(mpiprocs) + exec + mystr;
    end        
elseif pde.platform == "gpu"
    if mpiprocs==1        
        exec = "./gpuEXASIM ";     
        runstr = "!" + exec + mystr;
    else        
        exec = " ./gpumpiEXASIM ";
        runstr = "!" + mpirun + " -np " + string(mpiprocs) + exec + mystr;
    end
end

tic
eval(char(runstr));
toc

cd(char(cdir));

end

File: frontends/Julia/Gencode/runcode.jl
function runcode(pde, numpde=1)

display("Run C++ Exasim code ...")

cdir = pwd(); 
cd(pde.buildpath);

buildpath = pde.buildpath;
DataPath = buildpath;
mpirun = pde.mpirun;
pdenum = string(numpde) * " ";

if pde.platform == "cpu"        
    if pde.mpiprocs==1        
        str = "./cpuEXASIM " * pdenum * DataPath * "/datain/ " * DataPath * "/dataout/out";        
    else        
        str = mpirun * " -np " * string(pde.mpiprocs) * " ./cpumpiEXASIM " * pdenum * DataPath * "/datain/ " * DataPath * "/dataout/out";       
    end        
elseif pde.platform == "gpu"
    if pde.mpiprocs==1        
        str = "./gpuEXASIM " * pdenum * DataPath * "/datain/ " * DataPath * "/dataout/out";        
    else        
        str = mpirun * " -np " * string(pde.mpiprocs) * " ./gpumpiEXASIM " * pdenum * DataPath * "/datain/ " * DataPath * "/dataout/out";       
    end    
end

run(string2cmd(str));

cd(cdir);

return str;

end
